src/openrct2/actions/BannerPlaceAction.cpp

```diff
@@ -103,6 +103,7 @@ GameActions::Result BannerPlaceAction::Query() const
             GameActions::Status::InvalidParameters, STR_CANT_POSITION_THIS_HERE, STR_ERR_BANNER_ELEMENT_NOT_FOUND);
     }
     res.Cost = bannerEntry->price;
+    res.SetData(BannerPlaceActionResult{});
```

Comment:
You probably want to set the banner id here

Reply:
Actually not, for some reason the query does not produce a banner.

Reply:
My bad, I thought this was Execute, as for Query not having a banner id, you could pass the next possible id but it won't be guaranteed that the next banner will have that exact id so its a bit meaningless in that context.